### ISSUE TITLE
Feature: Improve the error message when attempting to install a library that cannot be found for specific Circuit Python version

### ISSUE THREAD

**Issue by mraleson** [mraleson/piku#21]
For example when neopixel package could not be found for a new CircuitPython version that isn't indexed:

```
piku add neopixel
Traceback (most recent call last):
  File "/site-packages/piku/main.py", line 71, in main
    args.cmd(args)
  File "/site-packages/piku/commands/add.py", line 43, in add_command
    previous, current, conflicts = add(package, constraint)
  File "/site-packages/piku/commands/add.py", line 7, in add
    package_version = packages.find(package_name, package_constraint)
  File "/site-packages/piku/core/packages.py", line 38, in find
    if package not in packages:
TypeError: argument of type 'NoneType' is not iterable
Failed.
```

**Comment by mraleson** [comment]
For example when neopixel package could not be found for a new CircuitPython version that isn't indexed:

```
piku add neopixel
Traceback (most recent call last):
  File "/site-packages/piku/main.py", line 71, in main
    args.cmd(args)
  File "/site-packages/piku/commands/add.py", line 43, in add_command
    previous, current, conflicts = add(package, constraint)
  File "/site-packages/piku/commands/add.py", line 7, in add
    package_version = packages.find(package_name, package_constraint)
  File "/site-packages/piku/core/packages.py", line 38, in find
    if package not in packages:
TypeError: argument of type 'NoneType' is not iterable
Failed.
```

**Comment by mraleson** [comment]
Closed with fedb3f6717b78da48afa1c1d02c18a665eb8267b
